### PR TITLE
add features and protected areas to proxy

### PR DESCRIPTION
**Pull request by elpamart**
Add protected areas and features endpoint in proxy.

### Overview

_Please write a description. If the PR is hard to understand, provide a quick explanation of the code._

### Designs

_Link to the related design prototypes (if applicable)_

### Testing instructions

_Please explain how to test the PR: ID of a dataset, steps to reach the feature, etc._

### Feature relevant tickets

_Link to the related task manager tickets_

---

## Checklist before submitting

- [ ] Meaningful commits and code rebased on `develop`.
- [ ] If this PR adds feature that should be tested for regressions when
      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
- [ ] Update CHANGELOG file

### CODE REVIEW

File: api/src/modules/proxy/proxy.controller.ts [hunk 1]
```diff
@@ -16,10 +16,9 @@ import { Request, Response } from 'express';
 
 @UseGuards(JwtAuthGuard)
 @ApiBearerAuth()
-@ApiTags('AdminArea vector tile proxy')
+@ApiTags('Vector tile proxy')
```
Comment (hotzevzl) [inline api/src/modules/proxy/proxy.controller.ts:19]: good point 😄 

if we add more/unrelated (to tiles) proxy endpoints in the future we may want to *then* split things across this and another controller, all using the same service and grouped under the same module, but just a thought for the future - maybe we won't need any of this 😅

File: api/src/modules/proxy/proxy.controller.ts [hunk 2]
```diff
@@ -67,13 +66,98 @@ export class ProxyController {
...
+    type: Number,
+    required: true,
+  })
+  @ApiQuery({
+    name: 'wdpaid',
+    description: 'Id of WDPA area',
+    type: Number,
+    required: false,
+    example: '5555697751',
+  })
```
Comment (hotzevzl) [inline api/src/modules/proxy/proxy.controller.ts:109]: which id are we after here? My understanding is that we may not have data in the `wdpa.wdpaid` column in the geodb for protected areas that are uploaded by users - should we use the `wdpa.id` UUID column instead? (moreover, right now `wdpaid` is nullable, whereas `id` is a PK)

File: api/src/modules/proxy/proxy.controller.ts [hunk 3]
```diff
@@ -67,13 +66,98 @@ export class ProxyController {
...
+    description: 'Specific id of the feature',
+    type: String,
+    required: true,
+  })
+  @ApiQuery({
+    name: 'extentId',
+    description: 'Extent of the project',
+    type: Number,
+    required: false,
+  })
```
Comment (hotzevzl) [inline api/src/modules/proxy/proxy.controller.ts:158]: is this correct? what does extentId map to (in our data model)?

File: geoprocessing/src/modules/protected-areas/protected-areas.controller.ts [hunk 4]
```diff
@@ -47,10 +47,10 @@ export class ProtectedAreasController<T> {
   })
   @ApiQuery({
     name: 'wdpaid',
-    description: 'Parent country of administrative areas in ISO code',
-    type: String,
+    description: 'Id of WDPA area',
```
Comment (hotzevzl) [inline geoprocessing/src/modules/protected-areas/protected-areas.controller.ts:50]: same as above in the corresponding controller in the API service

**Review by hotzevzl — COMMENTED** [review]
This is basically an approval of the PR. please see my main question about which IDs to use for protected areas and (if appropriate) update that - other than this, LGTM, thanks! 🚀

File: api/src/modules/proxy/proxy.controller.ts (hunk 1 above)
Comment (elpamart) [reply api/src/modules/proxy/proxy.controller.ts:19]: haha yes, I though we could add all the tiles endpoints under the same proxy and potentially split things if we add something not tile related.

File: api/src/modules/proxy/proxy.controller.ts (hunk 2 above)
Comment (elpamart) [reply api/src/modules/proxy/proxy.controller.ts:109]: that was @aagm suggestion and based on what we have in our endpoint definition doc. I guess if we are going to have the case of `wdpa.wdpaid` with no data for the areas uploaded by the user, I'm happy to change this param to `wdpa.id`.

@aagm what do you think?

File: api/src/modules/proxy/proxy.controller.ts (hunk 3 above)
Comment (elpamart) [reply api/src/modules/proxy/proxy.controller.ts:158]: yes, this is what we defined on the [endpoints definition doc](https://docs.google.com/document/d/1T1Gp9RM6J-CoZoyIL7XBeUU1_w2ZrvN7rCSN_XAnBSg/edit#). If I remember well was the` features.id` but I cannot check on my local as the features import didn't work. @aagm could confirm

File: api/src/modules/proxy/proxy.controller.ts (hunk 2 above)
Comment (aagm) [reply api/src/modules/proxy/proxy.controller.ts:109]: yes i was willing to use based on wdpaid, but i didn't though on user uploaded data, so changing it based on the db based id wdpa.id is ok from my point of view!

File: api/src/modules/proxy/proxy.controller.ts (hunk 3 above)
Comment (aagm) [reply api/src/modules/proxy/proxy.controller.ts:158]: This looks like a slip-on in my mind regarding how will we access the extent. I'm thinking if indeed we should refer to bbox here and filter by it.  But I think that based on the designs this is more related to the shape of the study area (or adm or extent, so I would refer it to a project id, and we can generate the specific logic from there.)

**Pull request merged**